llama-index-integrations/readers/llama-index-readers-azstorage-blob/llama_index/readers/azstorage_blob/base.py

```diff
@@ -42,34 +42,18 @@ class AzStorageBlobReader(BaseReader):
             The credentials with which to authenticate. This is optional if the account URL already has a SAS token.
     """
 
-    def __init__(
-        self,
-        *args: Any,
-        container_name: str,
-        prefix: str = "",
-        blob: Optional[str] = None,
-        name_starts_with: Optional[str] = None,
-        include: Optional[Any] = None,
-        file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = None,
-        connection_string: Optional[str] = None,
-        account_url: Optional[str] = None,
-        credential: Optional[Any] = None,
-        **kwargs: Any,
-    ) -> None:
-        """Initializes Azure Storage Account."""
-        super().__init__(*args, **kwargs)
-
-        self.container_name = container_name
-        self.prefix = prefix
-        self.connection_string = connection_string
-        self.blob = blob
-        self.name_starts_with = name_starts_with
-        self.include = include
-        self.file_extractor = file_extractor
-        self.account_url = account_url
-        self.credential = credential
-        # Not in use. As part of the TODO below. Is part of the kwargs.
-        # self.preloaded_data_path = kwargs.get('preloaded_data_path', None)
+    container_name: str
+    prefix: Optional[str] = ""
+    blob: Optional[str] = None
+    name_starts_with: Optional[str] = None
+    include: Optional[Any] = None
+    file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = None
```

Comment:
I think this should be ```
file_extractor: Optional[Dict[str, Union[str, BaseReader]]] = Field(
        default=None, exclude=True
    )
```
Because we want to exclude it from the serialization (since BaseReader isn't necessarily serializable)

Reply:
Also, I think it would be nice to add an unit test checking that both serialization and JSON schema generation work correctly. Look at https://github.com/run-llama/llama_index/pull/12061/files for reference

Reply:
Good call, updated